src/Symfony/Component/Console/Attribute/Input.php

```diff
@@ -0,0 +1,132 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Attribute\Reflection\ReflectionMember;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER | \Attribute::TARGET_PROPERTY)]
+final class Input
+{
+    /**
+     * @var array<string, Argument>
+     */
+    private array $arguments = [];
+
+    /**
+     * @var array<string, Option>
+     */
+    private array $options = [];
+
+    /**
+     * @var array<string, self>
+     */
+    private array $inputs = [];
+
+    private \ReflectionClass $class;
+
+    public static function tryFrom(\ReflectionParameter|\ReflectionProperty $member): ?self
+    {
+        $reflection = new ReflectionMember($member);
+
+        if (!$self = $reflection->getAttribute(self::class)) {
+            return null;
+        }
+
+        $type = $reflection->getType();
+
+        if (!$type instanceof \ReflectionNamedType) {
+            throw new LogicException(\sprintf('The input %s "%s" must have a named type.', $reflection->getMemberName(), $member->name));
+        }
+
+        if (!class_exists($class = $type->getName())) {
+            throw new LogicException(\sprintf('The input class "%s" does not exist.', $type->getName()));
+        }
+
+        $self->class = new \ReflectionClass($class);
+
+        foreach ($self->class->getProperties() as $property) {
+            if (!$property->isPublic() || $property->isStatic()) {
+                continue;
+            }
+
+            if ($argument = Argument::tryFrom($property)) {
+                $self->arguments[$property->name] = $argument;
+                continue;
+            }
+
+            if ($option = Option::tryFrom($property)) {
+                $self->options[$property->name] = $option;
+                continue;
+            }
+
+            if ($input = self::tryFrom($property)) {
+                $self->inputs[$property->name] = $input;
+            }
+        }
+
+        if (!$self->arguments && !$self->options && !$self->inputs) {
+            throw new LogicException(\sprintf('The input class "%s" must have at least one argument or option.', $self->class->name));
+        }
+
+        return $self;
+    }
+
+    /**
+     * @internal
+     */
+    public function resolveValue(InputInterface $input): mixed
+    {
+        $instance = $this->class->newInstanceWithoutConstructor();
+
+        foreach ($this->arguments as $name => $argument) {
+            $instance->$name = $argument->resolveValue($input);
+        }
+
+        foreach ($this->options as $name => $option) {
+            $instance->$name = $option->resolveValue($input);
+        }
+
+        foreach ($this->inputs as $name => $in) {
+            $instance->$name = $in->resolveValue($input);
+        }
+
+        return $instance;
+    }
+
+    /**
+     * @return Argument[]
```

Comment:
You can be more specific that this a list.
```suggestion
     * @return list<Argument>
```